functions/src/modules/SE.functions.ts

```diff
@@ -88,3 +93,276 @@ export const processNewAllotment = functions.database
+            modified: [] // Rows already found in the DB
+        };
+
+    
+    sheetscMetadata.forEach(async sheet => {
+        if (sheet.firstRow.indexOf('Beginning') < 0 || sheet.firstRow.indexOf('Ending') < 0) 
+            return;
+
+        let rows = await helpers.getSheetRows(sheet, sheets, spreadsheetId, sheet.firstRow);
+        
+        let chunks = {};
+        currentFile = { 
+            file_name: null,
+            chunks: [], 
+            skip: false, 
+            track: null 
+        };
+
+        ////////////////////////////////////////
+        //  
+        //  1. loop through all the rows in the current sheet
+        //  2. Apply the following checks and if something wrong is found
+        //      increment the corresonding error attribute in the `summary` object
+        //      [CHECKS]
+        //          A. Misplaced files
+        //          B. Non-existence of `Resolution` values in a row
+        //          C. Empty values in either `Beginning` or `Ending` attributes
+        //
+        ////////////////////////////////////////
+        for (const row of rows) {
+
+            let _import = validateSpreadsheetRow(row, summary, sheet.title);            
+            
+            const AudioFileName = row['Audio File Name'];
+            const continuationFrom = row['Continuation From'];
+
+            const chunkToStore = {
+                beginning: moment.duration(row['Beginning']).asSeconds(),
+                ending: moment.duration(row['Ending']).asSeconds(),
+                continuationFrom,
+                contentReporting: {
+                    date: row['Date'],
+                    locatioon: row['Location'],
+                    category: row['Category'],
+                    topics: row['Topics'],
+                    gurudevaTimings: row['Gurudeva Timings'],
+                    otherSpeakers: row['Other Speakers'],
+                    kirtan: row['Kirtan'],
+                    abruptLecture: row['Abrupt Lecture'],
+                    suggestedTitle: row['Suggested Title'],
+                    languages: row['Languages'],
+                    soundQualityRating: row['Sound Quality'],
+                    soundIssues: row['Sound Issues'],
+                    comments: row['Comments'],
+                    submissionTimestamp: row['Timestamp'],
+                    submissionSerial: row['Submission Serial'],
+                },
+                importTimestamp: admin.database.ServerValue.TIMESTAMP,
+                processingResolution: row['Resolution'],
+                _import
+            };
+
+
+            
+            // Same file name --> add newly read chunk (row) to the `currentFile` object
+            if (currentFile.file_name === AudioFileName) 
+                currentFile.chunks.push(chunkToStore);
+            
+            // NEW FILE --> Save old file AND add the newly read chunk to `currentFile` object            
```

Comment:
Some black magic is happening below.
Rather create another function that “flushes” the current file into the database.
The purpose of the `chunks` variable is unclear.

Reply:
The **chunks object** is used to group all the chunks (rows) of a single file ... 
The task card says the rows of an audio file are not guaranteed to be successive .. so I thought of using an object with the following structure

```javascript
chunks = {
    "BR-01A": [ chunk#1, chunk#2, .... ],
    "BR-03A": [ chunk#1, chunk#2, .... ],
}
```
and **after all the rows of a sheet are read**, the rows are flushed to the database.
That's why I can't flush the chunks of the **current file** whenever a row is read.

Do you suggest to flush the current file to the db whenever a new file name is encountered and later *append* any chunks that belong to a file **existing** in the db ?
@anantakrishna

Reply:
Bear in mind, the chunks have to be sorted by `Beginning` time, so pushing them to the db and rereading them for sorting would be a waste of time.

Reply:
> The task card says the rows of an audio file are not guaranteed to be successive

I should have described this in a more clear way, sorry. The card says:

- Rows related to a particular file name are supposed to be clubbed together.
- Rows within a file name group may be out of order.
- Rows are not guaranteed to be sorted by the file name.

This means that the chunks belonging to a particular file are grouped together, but these groups may be not sorted by the file name, and the chunks may be not sorted within a group. For example:

```
BR-03A chunk 1
BR-03A chunk 2
BR-01A chunk 3
BR-01A chunk 1
BR-01A chunk 2
```

So, we can “flush the current file to the db whenever a new file name is encountered”. And if later the same file name occurs, then an error should be reported and no overwriting should occur.